public/app/features/plugins/admin/state/selectors.ts

```diff
@@ -32,11 +33,15 @@ export type PluginFilters = {
 export const selectPlugins = (filters: PluginFilters) =>
   createSelector(selectAll, (plugins) => {
     const keyword = filters.keyword ? unEscapeStringFromRegex(filters.keyword.toLowerCase()) : '';
+    const pluginsForSearch: { [key: string]: string } = getPluginDetailsForFuzzySearch(plugins);
+    const pluginsAfterFuzzySearch = fuzzySearch(keyword, Object.values(pluginsForSearch));
```

Comment:
I agree with @xnyo that this line `const pluginsForSearch: { [key: string]: string } = getPluginDetailsForFuzzySearch(plugins);` is an implementation detail of the fuzzy search. 

Better if we leave this a simple one line API and you handle everything inside the fuzzy search method. We can also remove performance overhead of constructing all the fuzzysearch list and applying them when there's nothing to search. I suggest you use different clearer names for the variables and methods. e.g.:

```typescript
// let the `filterByKeyword` method handle all the creation and filtering by itsself. this function doesn't need to know how it does it.
const filteredPluginIds = keyword !== "" ? filterByKeyword(plugins, keyword) : null;
```